heartbeat/tests/system/test_autodiscovery.py

```diff
@@ -0,0 +1,63 @@
+import os
+from heartbeat import BaseTest
+import unittest
+import re
+
+from beat.beat import INTEGRATION_TESTS
+
+
+class TestAutodiscover(BaseTest):
+    """
+    Test heartbeat autodiscover
+    """
+    @unittest.skipIf(not INTEGRATION_TESTS or
+                     os.getenv("TESTING_ENVIRONMENT") == "2x",
+                     "integration test not available on 2.x")
+    def test_docker(self):
+        """
+        Test docker autodiscover starts modules from templates
+        """
+        import docker
+        docker_client = docker.from_env()
+
+        self.render_config_template(
+            autodiscover={
+                'docker': {
+                    'templates': '''
+                      - condition:
+                          equals.docker.container.image: redis:3.2.4
+                        config:
+                          - type: tcp
+                            hosts: ["${data.host}:${data.port}"]
+                            schedule: "@every 1s"
+                            timeout: 1s
+                    ''',
+                },
+            },
+        )
+
+        proc = self.start_beat()
+        docker_client.images.pull('redis:3.2.4')
```

Comment:
If we set TEST_ENVIRONMENT to true, these containers should already be running.

There are currently two models of testing against containers we have:

* Start everything at the beginning: Libbeat and others
* Start on request and stop afterwards: Metricbeat modules

We seem to mix the two here. For simplicity I would start with setting TEST_ENVIRONMENT to true and not handle container start / stop in here.